desktop/app/main-window.js

```diff
@@ -33,15 +33,18 @@ export default function () {
     mainWindow.window.setPosition(forceMainWindowPosition.x, forceMainWindowPosition.y)
   }
 
+  const openedAtLogin = app.getLoginItemSettings().openAtLogin
   const isRestore = getenv.boolish('KEYBASE_RESTORE_UI', false) || app.getLoginItemSettings().restoreState
   const openHidden = (getenv.string('KEYBASE_START_UI', '') === 'hideWindow') || app.getLoginItemSettings().wasOpenedAsHidden
 
-  // We show the main window on startup if:
-  //  - We are restoring UI and the window was previously visible (in app state)
-  //  - and, we are not set to open hidden
-  const showMainWindow = (isRestore && !appState.state.windowHidden) && !openHidden
-  console.log('Show main window: %s', showMainWindow)
-  if (showMainWindow) {
+  // Don't show main window:
+  // - If we are set to open hidden,
+  // - or, if we are restoring and window was hidden
+  // - or, if we are opening from login (but not restoring)
+  const hideMainWindow = openHidden || (isRestore && appState.state.windowHidden) || (openedAtLogin && !isRestore)
```

Comment:
I'm reading `openedAtLogin && !isRestore` as "the app is set to open at login and we're not launching in restore mode". Is this not true when the user clicks the dock icon, if `openAtLogin` is set? Does the main window show up when this is the case?